File: src/schemas/integration.ts
import { Schema, model, Types } from 'mongoose';

export type Integration = {
  _id: Types.ObjectId;
  accessToken: string;
  refreshToken: string;
  ownerId: string;
  avatar: string;
  integrationId: string;
  username: string;
  type: string;
  expires: number;
};

export const IntegrationModel = model(
  'Integration',
  new Schema({
    accessToken: String,
    refreshToken: String,
    ownerId: String,
    avatar: String,
    integrationId: String,
    username: String,
    type: String,
    expires: Number,
  }),
);

File: src/main/java/com/staroverlay/eventsub/entities/Integration.java
package com.staroverlay.eventsub.entities;

import com.sammwy.classserializer.annotations.Prop;
import com.sammwy.milkshake.Entity;

public class Integration extends Entity {
  @Prop
  public String accessToken;

  @Prop
  public String refreshToken;

  @Prop
  public String ownerId;

  @Prop
  public String avatar;

  @Prop
  public String integrationId;

  @Prop
  public String username;

  @Prop
  public String type;

  @Prop
  public long expires;
}
